test/e2e/storage/sanity.go

```diff
@@ -995,3 +996,40 @@ func WaitForPodsWithLabelRunningReady(c clientset.Interface, ns string, label la
 		})
 	return pods, err
 }
+
+// Register list of volumes before test, using out-of-band host commands (i.e. not CSI API).
+// checkForLeftovers lists volumes again after test, diff means leftovers.
+// TODO: remove debug prints before merging
+func getHostVolumes(numWorkers int) map[string][]string {
+	var cmd string
+	var hdr string
+	//var result map[string][]string
+	switch os.Getenv("TEST_DEVICEMODE") {
+	case "lvm":
+		// lvs adds many space (0x20) chars at end, squeeze repetitions to one
+		cmd = "sudo lvs --foreign --noheadings |tr -s ' '"
+		hdr = "LVM Volumes"
+	case "direct":
+		// ndctl produces multiline block. We want one line per namespace.
+		// Remove newlines, then insert one at the end, clean some more.
+		cmd = "sudo ndctl list |tr -d '\n' |tr ']' '\n' |tr -d '[{}\"' |tr -d ' '"
+		hdr = "Namespaces"
+	}
+	result := make(map[string][]string)
+	//framework.Logf("getHostVolumes-DEBUG: cmd:[%s]", cmd)
+	for worker := 1; worker <= numWorkers; worker++ {
+		sshcmd := fmt.Sprintf("%s/_work/%s/ssh.%d", os.Getenv("REPO_ROOT"), os.Getenv("CLUSTER"), worker)
+		ssh := exec.Command(sshcmd, cmd)
+		out, _ := ssh.CombinedOutput()
+		buf := fmt.Sprintf("%s on Node %d", hdr, worker)
+		result[buf] = strings.Split(string(out), "\n")
```

Comment:
I think if you trim the output before splitting, then we'll get rid of one redundant empty entry.

Reply:
Thanks, indeed TrimSpace takes away that extra trailing newline.
Pushed new version with that change, also added a separate tmp-commit where I try to modify Jenkinsfile not to stop after LVM stage failure